plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesByLatencyService.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.search.SearchType;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.plugin.insights.core.exporter.QueryInsightLocalIndexExporter;
+import org.opensearch.plugin.insights.rules.model.SearchQueryLatencyRecord;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.PriorityBlockingQueue;
+
+/**
+ * Service responsible for gathering, analyzing, storing and exporting
+ * top N queries with high latency data for search queries
+ *
+ * @opensearch.internal
+ */
+public class TopQueriesByLatencyService extends QueryInsightService<
+    SearchQueryLatencyRecord,
+    PriorityBlockingQueue<SearchQueryLatencyRecord>,
+    QueryInsightLocalIndexExporter<SearchQueryLatencyRecord>> {
+    private static final Logger log = LogManager.getLogger(TopQueriesByLatencyService.class);
+
+    /** Default window size in seconds to keep the top N queries with latency data in query insight store */
+    public static final int DEFAULT_WINDOW_SIZE = 60;
```

Comment:
We should probably have const variables for window size and top N size "upper limit", based on benchmarking results.